packages/ui-components/src/components/search/search.tsx

```diff
@@ -0,0 +1,69 @@
+import { Component, h, Event, EventEmitter, Prop, Host, State, Watch } from '@stencil/core';
+import { isEmpty } from 'lodash-es';
+import { EComponentSize } from '../../types';
+import { EInputFieldType } from '../text-field/text-field.types';
+import { ISearchEvents } from './search.types';
+
+@Component({
+	tag: 'kv-search',
+	shadow: true
+})
+export class KvSearch implements ISearchEvents {
+	/** (optional) Search place holder */
+	@Prop({ reflect: true }) placeholder: string = 'Search';
+	/** (optional) Sets this tab item to a different styling configuration */
+	@Prop() size?: EComponentSize = EComponentSize.Large;
+	/** (optional) Search disabled */
+	@Prop({ reflect: true }) disabled = false;
+	/** Search value*/
+	@Prop({ reflect: true }) value?: string;
+	/** @inheritdoc */
+	@Event() textChange: EventEmitter<string>;
+	/** @inheritdoc */
+	@Event() textFieldBlur: EventEmitter<string>;
+	/** @inheritdoc */
+	@Event() clickResetButton: EventEmitter<MouseEvent>;
+	/** Internal show reset button state */
+	@State() showResetIcon = !isEmpty(this.value);
+	/** Internal value state */
+	@State() _value: string = this.value;
+	/** Watch changes on value to ensure reset event clears the text */
+	@Watch('value')
+	valueChangeHandler(newValue: string) {
+		this._value = newValue;
+		this.showResetIcon = !isEmpty(newValue);
+	}
+
+	/** Icons */
+	private searchIcon = 'kv-search';
+	private resetIcon = 'kv-close';
+
+	private onTextChange = (event: CustomEvent<string>) => {
+		this._value = event.detail;
+		this.showResetIcon = !isEmpty(this._value);
+	};
+
+	private onResetClick = (event: MouseEvent) => {
+		this.showResetIcon = false;
```

Comment:
This value gonna change when the user clean the result (if they want it) if not we gonna have a state problem since the X will disappear but the text value continue with the same data.

Reply:
The button now appears if the value has any text on it. The value itself is updated from the parent component of the search via prop. 
The state showResetIcon was removed.